File: ProTip365/Detail/DetailLocalization.swift
import SwiftUI

struct DetailLocalization {
    let language: String

    init(language: String) {
        self.language = language
    }

    var noDataText: String {
        switch language {
        case "fr": return "Aucune donnée"
        case "es": return "Sin datos"
        default: return "No data"
        }
    }

    var doneText: String {
        switch language {
        case "fr": return "Terminé"
        case "es": return "Listo"
        default: return "Done"
        }
    }

    var salesText: String {
        switch language {
        case "fr": return "Ventes"
        case "es": return "Ventas"
        default: return "Sales"
        }
    }

    var tipsText: String {
        switch language {
        case "fr": return "Pourboires"
        case "es": return "Propinas"
        default: return "Tips"
        }
    }

    var detailsText: String {
        switch language {
        case "fr": return "Détails"
        case "es": return "Detalles"
        default: return "Details"
        }
    }

    var incomeBreakdownText: String {
        switch language {
        case "fr": return "RÉPARTITION DES REVENUS"
        case "es": return "DESGLOSE DE INGRESOS"
        default: return "INCOME BREAKDOWN"
        }
    }

    var performanceMetricsText: String {
        switch language {
        case "fr": return "MÉTRIQUES DE PERFORMANCE"
        case "es": return "MÉTRICAS DE RENDIMIENTO"
        default: return "PERFORMANCE METRICS"
        }
    }

    var shiftDetailsText: String {
        switch language {
        case "fr": return "DÉTAILS DES QUARTS"
        case "es": return "DETALLES DE TURNOS"
        default: return "SHIFT DETAILS"
        }
    }

    var salaryText: String {
        switch language {
        case "fr": return "Salaire"
        case "es": return "Salario"
        default: return "Salary"
        }
    }

    var grossSalaryText: String {
        switch language {
        case "fr": return "Salaire brut"
        case "es": return "Salario bruto"
        default: return "Gross Salary"
        }
    }

    var netSalaryText: String {
        switch language {
        case "fr": return "Salaire net"
        case "es": return "Salario neto"
        default: return "Net Salary"
        }
    }

    var hoursText: String {
        switch language {
        case "fr": return "Heures"
        case "es": return "Horas"
        default: return "Hours"
        }
    }

    var totalText: String {
        switch language {
        case "fr": return "Total"
        case "es": return "Total"
        default: return "Total"
        }
    }

    var otherText: String {
        switch language {
        case "fr": return "Autre"
        case "es": return "Otro"
        default: return "Other"
        }
    }

    var tipOutText: String {
        switch language {
        case "fr": return "Partage"
        case "es": return "Reparto"
        default: return "Tip Out"
        }
    }

    var totalIncomeText: String {
        switch language {
        case "fr": return "Revenu total"
        case "es": return "Ingresos totales"
        default: return "Total Income"
        }
    }

    var totalHoursText: String {
        switch language {
        case "fr": return "Heures totales"
        case "es": return "Horas totales"
        default: return "Total Hours"
        }
    }

    var totalSalesText: String {
        switch language {
        case "fr": return "Ventes totales"
        case "es": return "Ventas totales"
        default: return "Total Sales"
        }
    }

    var tipPercentText: String {
        switch language {
        case "fr": return "% Pourboire"
        case "es": return "% Propina"
        default: return "Tip %"
        }
    }

    var targetText: String {
        switch language {
        case "fr": return "Objectif"
        case "es": return "Objetivo"
        default: return "Target"
        }
    }

    var noShiftsRecordedText: String {
        switch language {
        case "fr": return "Aucun quart enregistré pour cette période"
        case "es": return "No hay turnos registrados para este período"
        default: return "No shifts recorded for this period"
        }
    }

    var incomeText: String {
        switch language {
        case "fr": return "Revenu"
        case "es": return "Ingresos"
        default: return "Income"
        }
    }

    var summaryText: String {
        switch language {
        case "fr": return "Résumé"
        case "es": return "Resumen"
        default: return "Summary"
        }
    }

    func detailTitle(for detailType: String) -> String {
        switch detailType {
        case "income": return incomeText
        case "tips": return tipsText
        case "sales": return salesText
        case "hours": return hoursText
        case "total": return summaryText
        case "other": return otherText
        default: return detailsText
        }
    }
}

File: ProTip365Android/app/src/main/java/com/protip365/app/presentation/detail/DetailLocalization.kt
package com.protip365.app.presentation.detail

class DetailLocalization(private val language: String) {
    
    val noDataText: String
        get() = when (language) {
            "fr" -> "Aucune donnée"
            "es" -> "Sin datos"
            else -> "No data"
        }
    
    val doneText: String
        get() = when (language) {
            "fr" -> "Terminé"
            "es" -> "Listo"
            else -> "Done"
        }
    
    val salesText: String
        get() = when (language) {
            "fr" -> "Ventes"
            "es" -> "Ventas"
            else -> "Sales"
        }
    
    val tipsText: String
        get() = when (language) {
            "fr" -> "Pourboires"
            "es" -> "Propinas"
            else -> "Tips"
        }
    
    val detailsText: String
        get() = when (language) {
            "fr" -> "Détails"
            "es" -> "Detalles"
            else -> "Details"
        }
    
    val incomeBreakdownText: String
        get() = when (language) {
            "fr" -> "RÉPARTITION DES REVENUS"
            "es" -> "DESGLOSE DE INGRESOS"
            else -> "INCOME BREAKDOWN"
        }
    
    val performanceMetricsText: String
        get() = when (language) {
            "fr" -> "MÉTRIQUES DE PERFORMANCE"
            "es" -> "MÉTRICAS DE RENDIMIENTO"
            else -> "PERFORMANCE METRICS"
        }
    
    val shiftDetailsText: String
        get() = when (language) {
            "fr" -> "DÉTAILS DES QUARTS"
            "es" -> "DETALLES DE TURNOS"
            else -> "SHIFT DETAILS"
        }
    
    val salaryText: String
        get() = when (language) {
            "fr" -> "Salaire"
            "es" -> "Salario"
            else -> "Salary"
        }
    
    val grossSalaryText: String
        get() = when (language) {
            "fr" -> "Salaire brut"
            "es" -> "Salario bruto"
            else -> "Gross Salary"
        }
    
    val netSalaryText: String
        get() = when (language) {
            "fr" -> "Salaire net"
            "es" -> "Salario neto"
            else -> "Net Salary"
        }
    
    val hoursText: String
        get() = when (language) {
            "fr" -> "Heures"
            "es" -> "Horas"
            else -> "Hours"
        }
    
    val totalText: String
        get() = when (language) {
            "fr" -> "Total"
            "es" -> "Total"
            else -> "Total"
        }
    
    val otherText: String
        get() = when (language) {
            "fr" -> "Autre"
            "es" -> "Otro"
            else -> "Other"
        }
    
    val tipOutText: String
        get() = when (language) {
            "fr" -> "Partage"
            "es" -> "Reparto"
            else -> "Tip Out"
        }
    
    val totalIncomeText: String
        get() = when (language) {
            "fr" -> "Revenu total"
            "es" -> "Ingresos totales"
            else -> "Total Income"
        }
    
    val totalHoursText: String
        get() = when (language) {
            "fr" -> "Heures totales"
            "es" -> "Horas totales"
            else -> "Total Hours"
        }
    
    val totalSalesText: String
        get() = when (language) {
            "fr" -> "Ventes totales"
            "es" -> "Ventas totales"
            else -> "Total Sales"
        }
    
    val tipPercentText: String
        get() = when (language) {
            "fr" -> "% Pourboire"
            "es" -> "% Propina"
            else -> "Tip %"
        }
    
    val targetText: String
        get() = when (language) {
            "fr" -> "Objectif"
            "es" -> "Objetivo"
            else -> "Target"
        }
    
    val noShiftsRecordedText: String
        get() = when (language) {
            "fr" -> "Aucun quart enregistré pour cette période"
            "es" -> "No hay turnos registrados para este período"
            else -> "No shifts recorded for this period"
        }
    
    val incomeText: String
        get() = when (language) {
            "fr" -> "Revenu"
            "es" -> "Ingresos"
            else -> "Income"
        }
    
    val summaryText: String
        get() = when (language) {
            "fr" -> "Résumé"
            "es" -> "Resumen"
            else -> "Summary"
        }
    
    val avgHourlyRateText: String
        get() = when (language) {
            "fr" -> "Taux horaire moyen"
            "es" -> "Tarifa horaria promedio"
            else -> "Avg Hourly Rate"
        }
    
    val shiftsText: String
        get() = when (language) {
            "fr" -> "quarts"
            "es" -> "turnos"
            else -> "shifts"
        }
    
    val shiftText: String
        get() = when (language) {
            "fr" -> "quart"
            "es" -> "turno"
            else -> "shift"
        }
    
    fun detailTitle(detailType: String): String {
        return when (detailType) {
            "income" -> incomeText
            "tips" -> tipsText
            "sales" -> salesText
            "hours" -> hoursText
            "total" -> summaryText
            "other" -> otherText
            else -> detailsText
        }
    }
    
    fun getPeriodText(period: String): String {
        return when (period.lowercase()) {
            "today" -> when (language) {
                "fr" -> "Aujourd'hui"
                "es" -> "Hoy"
                else -> "Today"
            }
            "week" -> when (language) {
                "fr" -> "Semaine"
                "es" -> "Semana"
                else -> "Week"
            }
            "month" -> when (language) {
                "fr" -> "Mois"
                "es" -> "Mes"
                else -> "Month"
            }
            "year" -> when (language) {
                "fr" -> "Année"
                "es" -> "Año"
                else -> "Year"
            }
            else -> period
        }
    }
}

